docs/dub-reference/subpackages.md

```diff
@@ -6,7 +6,130 @@
+    expected to only be used within your project.
+
+## Usage
+
+The syntax used to reference sub-packages in commands is often `<package>:<sub-package>` or just `:<sub-package>`.
+
+- To only add this sub-package from a package as a dependency, execute `dub add <packagename>:<sub-package>`
+- To build a sub-package, execute `dub build :<sub-package>`
+- To build and run a sub-package, execute `dub run :<sub-package>`
+- To test a sub-package, execute `dub test :<sub-package>`
+
+When the package name is omitted, the current working package name is assumed (e.g., `<package>`).
+
+Because sub-packages act as any other package, the default `targetType` is `autodetect`.
+
+When a sub-package is built, additional "Have" versions are defined.
+For example, with a root package named `package` with a `mylibrary` sub-package,
+a `Have_package_mylibrary` version will be automatically defined by DUB,
+and will be visible for all packages.
+
+## Declaration
+
+Sub-packages can be declared entirely within the main recipe, or by a
+reference to a name of a folder containing the sub-package recipe file.
+
+If you have a subcomponent called "mylibrary", it can simply be referenced by name.
+
+To be useful, a sub-package needs, at minimum, a name.
+
+=== "dub.sdl"
+
+    ```sdl
+    subPackage "mylibrary"
+    ```
+
+=== "dub.json"
+
+    ```json
+    {
+        "subPackages": [ "mylibrary" ]
+    }
+    ```
+
+!!! note
+
+    While the root recipe can contain sub-package configurations, it is
+    preferenced to leave configurations in the sub-package recipe file,
+    for ease of management.
+
+## Example
+
+This minimal working example assumes this file structure:
+
+- `dub.sdl` (or `dub.json`): Main recipe.
+- `source/` folder with:
+    - `app.d`: Main source. Needs to import our sub-package library as `import mylibrary;`.
+- `mylibrary/` folder with:
+    - `dub.sdl` (or `dub.json`): Library recipe.
+    - `source/` folder with:
+        - `example.d`: Library source. Module needs to be declared as `module mylibrary;`.
+
+!!! note
+
+    If you plan to publish your package to the main registry, avoid naming
+    sub-packages "example", as it causes management and discovery issues.
+
+The main recipe file can reference the sub-package.
+
+=== "dub.sdl"
+
+    ```sdl
+    name "dubtest"
+    dependency ":mylibrary" version="*"
+    subPackage "mylibrary"
+    ```
+
+=== "dub.json"
+
+    ```json
+    {
+        "name": "dubtest",
+        "dependencies": { ":mylibrary": "*" },
+        "subPackages": [ "mylibrary" ]
+    }
+    ```
+
+The sub-package recipe file can have its own set of dedicated configurations,
+but only needs a name to be usable.
+
+=== "mylibrary/dub.sdl"
+
+    ```sdl
+    name "mylibrary"
+    ```
+
+=== "mylibrary/dub.json"
+
+    ```json
+    {
+        "name": "mylibrary"
+    }
+    ```
+
+The `mylibrary` sub-package can now be tested with `dub test --root=mylibrary` from the
```

Comment:
```suggestion
The `mylibrary` sub-package can now be tested with `dub test :mylibrary` from the
```

the `--root` can be used for packages that are not sub-packages, but just standalone packages within the repository folder, which aren't published to the registry